dask/array/core.py

```diff
@@ -787,8 +791,18 @@ def store(sources, targets, lock=True, compute=True, **kwargs):
         raise ValueError("Different number of sources [%d] and targets [%d]"
                          % (len(sources), len(targets)))
 
-    updates = [insert_to_ooc(tgt, src, lock=lock)
-               for tgt, src in zip(targets, sources)]
+    if isinstance(regions, tuple) or regions is None:
+        regions = [regions]
+
+    if len(sources) > 1 and len(regions) == 1:
+        regions = [regions[0] for src in sources]
```

Comment:
A clearer way to express this is `regions = regions * len(source)` or even `regions *= len(sources)`